src/plugins/interaction/GraphInteraction/GraphInteraction.cpp

```diff
@@ -77,112 +88,122 @@ std::ifstream& GotoLine(std::ifstream& file, unsigned int num) {
-        std::map<int, Eigen::Vector3d> nodes;
-        auto graph_msg = std::make_shared<sc::Message<sm::Graph>>();
-        graph_msg->data.set_id(id_);
-        unsigned int edge_counter = 0;
-        while (std::getline(graph_file, line)) {
-            std::vector<std::string> words;
-            boost::split(words, line, [](char c) { return c == ' '; });
-
-            bool is_node = words.size() == 3;
-            if (is_node) {
-                double x, y, z;
-                int id = std::stoi(words[0]);
-                double latitude = std::stod(words[1]), longitude = std::stod(words[2]);
-                parent_->projection()->Forward(latitude, longitude, 0.0, x, y, z);
-                nodes[id] = Eigen::Vector3d(x, y, z);
-
-                auto node_ptr = graph_msg->data.add_nodes();
-                node_ptr->set_id(id);
-                sc::set(node_ptr->mutable_point(), nodes[id]);
-            } else { // is edge
-                edge_counter++;
-                int id_start = std::stoi(words[0]), id_end = std::stoi(words[1]);
-                double length = std::stod(words[2]);
-
-                bool edge_nodes_exist = nodes.count(id_start) > 0 && nodes.count(id_end) > 0; // reality check
-                if (edge_nodes_exist) {
-                    auto edge_ptr = graph_msg->data.add_edges();
-                    edge_ptr->set_start_node_id(id_start);
-                    edge_ptr->set_end_node_id(id_end);
-                    edge_ptr->set_weight(length);
-
-                    if (found_labels) {
-                        // Get Street Name
-                        GotoLine(names_file, edge_counter);
-                        std::string street_name;
-                        std::getline(names_file, street_name);
-                        edge_ptr->set_label(street_name);
-                    }
-
-                    // Visualize the Edges
-                    if (vis_graph_) {
-                        auto edge_shape = std::make_shared<scrimmage_proto::Shape>();
-                        edge_shape->set_persistent(true);
-                        edge_shape->set_opacity(1.0);
-                        scrimmage::set(edge_shape->mutable_color(), 255, 0, 0);
-                        scrimmage::set(edge_shape->mutable_line()->mutable_start(), nodes[id_start]);
-                        scrimmage::set(edge_shape->mutable_line()->mutable_end(), nodes[id_end]);
-                        draw_shape(edge_shape);
-                    }
-                }
-            }
-        }
-        pub_graph_->publish(graph_msg);
+    std::pair<boost::adjacency_list<>::vertex_iterator,
+        boost::adjacency_list<>::vertex_iterator> vs = boost::vertices(g_);
+
+    std::map<uint64_t, Eigen::Vector3d> nodes;
+    std::map<uint64_t, uint64_t> boost_vert_to_osmid;
+
+    auto graph_msg = std::make_shared<sc::Message<sm::Graph>>();
+    graph_msg->data.set_id(id_);
+    for (; vs.first != vs.second; ++vs.first) {
```

Comment:
can we just initialize this variable in the loop? In other words, `auto vs = boost::vertices(g_); vs.first != vs.second; ++vs.first`.

Reply:
Yes, good idea. Thanks!